commit 56d2ac763829d2443075e8266dd00166ee11c80d
Author: Bob Copeland <me@bobcopeland.com>
Date:   Wed Apr 15 07:57:33 2009 -0400

    ath5k: use tasklet_hi_schedule for beacon queue
    
    For embedded platforms, beacon transmission can be starved when
    flooded with data packets.  Prioritize beacons by giving the beacon
    queue the first shot when the isr completes.
    
    Changes-licensed-under: 3-Clause-BSD
    
    Signed-off-by: Bob Copeland <me@bobcopeland.com>
    Acked-by: Nick Kossifidis <mickflemm@gmail.com>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

diff --git a/drivers/net/wireless/ath5k/base.c b/drivers/net/wireless/ath5k/base.c
index ff6d4f8..ef8523e 100644
--- a/drivers/net/wireless/ath5k/base.c
+++ b/drivers/net/wireless/ath5k/base.c
@@ -2496,7 +2496,7 @@ ath5k_intr(int irq, void *dev_id)
 			tasklet_schedule(&sc->restq);
 		} else {
 			if (status & AR5K_INT_SWBA) {
-				tasklet_schedule(&sc->beacontq);
+				tasklet_hi_schedule(&sc->beacontq);
 			}
 			if (status & AR5K_INT_RXEOL) {
 				/*
